Feature/post blitz catalog

andrewpmiller opened the pull request:
## What this PR does:

- Adds a catalog page that is more suitable for post-blitz life
- Preview here: ~https://deploy-preview-248--thegymcms.netlify.com/static/take5/catalog~ https://deploy-preview-248--thegymcms.netlify.com/static/take5

### Important Notes

- Before submitting for review, turn off the debug view by uncommenting the conditional head/foot includes, and deleting the test include directives

andrewpmiller committed:
apply blitz style to topic-grouped catalog

andrewpmiller committed:
style topic grouper include

andrewpmiller committed:
force debug mode for testing the PR

andrewpmiller committed:
fix the debug view

andrewpmiller committed:
remove topic subtitle (redundant)

jgagne committed:
Update catalog structure and indentation

- Update structure; add end div tags and remove a stray end section tag
- Remove “Stay Up-To-Date” content
- Tidy up indentation

jgagne committed:
Update catalog view styles

- Add catalog related classes
- Improve card layout and style
- Improve vertical offset of sections for jump link to fragment navigation
- Tighten up heading letter-spacing

jgagne committed:
Update featured section

- Add card layout styles
- Update structure order of recent/topics sections to topics (navigation first and recent (featured content) second
- Add styles to update order of topics/recent sections; ltr (topics/recent) in single-column layout view and rtl (recent/topics) in multi-column layout view
- Update font-weight for topics subhead
- Remove !important declaration
- Remove ID selector lead-in
- Add consistent class to style instructor/description content
- Add featured specific instructor style

jgagne committed:
Update topics section

- Add navigation semantics
- Add navigation styles
- Update height of faux heading to match height of legit heading
- Update “details-opener” script to select topics disclosure menu
- Prepend “Browse” to “Topics” summary text in single-column layout view

jgagne committed:
Update disclosure widget styles

- Update upcoming/resources/transcript section style for single-column layout view
- Update list vertical spacing
- Update list-item line-height for better readability

jgagne committed:
Update catalog description

Less is more.

jgagne committed:
Update transcript list styles

- Update list vertical spacing
- Update list-item font-size/line-height

andrewpmiller committed:
update debug message for accuracy

andrewpmiller commented:
@jgagne — for some reason the topics aren't visible when the page is rendering. 

The generated page source for that section shows the topic nav items:

```html
<div class="col-md-4">
<nav class="take5--tutorial-topics" aria-labelledby="tutorial-topics">
    <details class="take5--topics-menu" id="tutorial-topics">
    <summary aria-expanded="true/false" tabindex="0" role="button">
        <b role="heading">Topics</b>
    </summary>
    <div class="take5--topics-menu-list">
        <!-- Found 3 topic(s) with live content -->
        <ul>
        
        <li>
            <a href="#web-design-development"><b>Web Design & Development</b></a>
        </li>
        
        <li>
            <a href="#prototyping"><b>Prototyping</b></a>
        </li>
        
        <li>
            <a href="#career-skills"><b>Career Skills</b></a>
        </li>
        
        </ul>
    </div>
    </details>
</nav>
</div>
```
The rendered screen does not: 

![image](https://user-images.githubusercontent.com/5747506/69445756-fa704e00-0d20-11ea-9b73-1acb8a715632.png)

Make sure to re-pull this branch before delving into it... I've updated the debug (commented) message to report the actual number of topics found)

jgagne commented:
@andrewpmiller a quick patch for the topics section is to add the `open` attribute to the details element, which will display the section open on all screen widths. For example: `<details open class="take5--topics-menu" id="tutorial-topics">`. I’ll work on a proper fix. To be continued, but adding the `open` attribute for now will make the content accessible.

jgagne committed:
Update disclosure widget behavior

- Remove erroneous duplicate lines from script
- Add open attribute to details element (content first with no JavaScript)

jgagne committed:
Cleanup CSS

- Remove unused comment blocks
- Fix incorrect value

jgagne committed:
Update featured section layout

- Update styles to reorder topics/recent sections rtl (recent/topics) in multi-column layout view
- Remove unwanted padding
- Update vertical padding based on layout
- Update title vertical alignment
- Fix disclosure widget styles

andrewpmiller commented:
@jgagne - I think this is looking great. I’ll get it ready for review so we can get it staged and ready for production.

andrewpmiller committed:
prep for review

andrewpmiller commented:
@mbifulco — this PR is ready for review whenever you've got a moment. Let me know if you run into conflicts, I'm happy to help sort them out.

andrewpmiller committed:
Merge branch 'staging' into feature/post-blitz-catalog

andrewpmiller committed:
change paths so new catalog appears

mbifulco commented on css/take5.css:
```diff
@@ -3,7 +3,7 @@
 #content > .container > .row,
 #content > .container > .row > .col-md-12 {
-align-content:stretch;align-items:stretch;align-self:auto;animation-delay:0s;animation-direction:normal;animation-duration:0s;animation-fill-mode:none;animation-iteration-count:1;animation-name:none;animation-play-state:running;animation-timing-function:ease;backface-visibility:visible;background-blend-mode:normal;block-size:auto;border-block-end-color:currentcolor;border-block-end-style:none;bor ...
+align-content:stretch;align-items:stretch;align-self:auto;animation-delay:0s;animation-direction:normal;animation-duration:0s;animation-fill-mode:none;animation-iteration-count:1;animation-name:none;animation-play-state:running;animation-timing-function:ease;backface-visibility:visible;background-blend-mode:normal;block-size:auto;border-block-end-color:currentcolor;border-block-end-style:none;bor ...
```
👀 what's going on here? CSS reset?

andrewpmiller replied:
Uh... that’s a questionbest answered by @jgagne I think

jgagne replied:
This is (or was) a reset for bootstrap flavored structure (within the page template) outside of the include, carried over (that applied) from designing the UX Design landing page. However, if it no longer applies, we can omit.

mbifulco reviewed (approved):
Looks like it's working from my end - pending the one question, let's get this thing into the real world 👍
